driver/src/price_estimation/dexag/api.rs

```diff
@@ -55,12 +55,15 @@ impl DexagHttpApi {
 }
 
 impl DexagApi for DexagHttpApi {
-    fn get_token_list(&self) -> Result<Vec<Token>> {
-        let mut url = self.base_url.clone();
-        url.set_path("token-list-full");
-        self.client
-            .get_json(url.to_string(), HttpLabel::Dexag)
-            .context("failed to parse token list json from dexag response")
+    fn get_token_list<'a>(&'a self) -> BoxFuture<'a, Result<Vec<Token>>> {
+        async move {
+            let mut url = self.base_url.clone();
+            url.set_path("token-list-full");
+            self.client
+                .get_json(url.to_string(), HttpLabel::Dexag)
```

Comment:
Should we use `get_json_async().await?` here so that we can yield while waiting on the response?

Reply:
Definitely! I missed that.

One thing we should do, once all the HTTP components are migrated, is remove the blocking requests so that we ensure that they aren't accidentally being used anywhere.

Reply:
Ah I missed that too. I should have grepped for this string.